### PR TITLE
ddl: support `DROP PLACEMENT` clause

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

It is a part of the project `Define Placement Rules in SQL`.

It is just simple. Dropping is essentially same as the half part of alter statement.

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Start pd-server, tikv-server, tidb-server(PD is excepted to have newer version than a26d9cf), enable placement rules using pd-ctl.

1. `alter table employees alter partition p0 add placement policy constraints='{"+zone=sh":1, "-zone=bj": 2}' role=voter replicas=4`. Except three more rules using pd-ctl config placement-rules show.
2. `alter table employees alter partition p0 drop placement policy role=voter` new rules are dropped.

### Release note

- support `DROP PLACEMENT` clause

### CODE REVIEW

File: ddl/placement_rule_test.go [hunk 1]
```diff
@@ -194,6 +194,30 @@ func (s *testPlacementSuite) TestPlacementBuild(c *C) {
...
+					Role:        ast.PlacementRoleVoter,
+					Tp:          ast.PlacementAdd,
+					Replicas:    3,
+					Constraints: `["+  zone=sh", "-zone = bj"]`,
+				},
+				{
+					Role: ast.PlacementRoleVoter,
+					Tp:   ast.PlacementDrop,
+				},
+			},
```
Comment (djshow832) [inline ddl/placement_rule_test.go:209]: Also try dropping a different role with Add or Alter.

Comment (djshow832) [reply ddl/placement_rule_test.go:209]: And multiple DROPs.

Comment (xhebox) [reply ddl/placement_rule_test.go:214]: two more tests here, and some more tests in sql. A better error output also.

File: ddl/placement_rule_test.go [hunk 2]
```diff
@@ -194,13 +198,107 @@ func (s *testPlacementSuite) TestPlacementBuild(c *C) {
...
+						},
+					},
+				},
+			},
+		},
 	}
 	for k, t := range tests {
 		out, err := buildPlacementSpecs(t.input)
-		c.Logf("test %d\n", k)
+		c.Logf("test %d", k)
```
Comment (djshow832) [inline ddl/placement_rule_test.go:298]: Some of the logs can be removed. Only log it when the result is not expected.

Comment (xhebox) [reply ddl/placement_rule_test.go]: it is moved into the fail block.

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -5252,7 +5252,7 @@ func buildPlacementSpecs(specs []*ast.PlacementSpec) ([]*placement.RuleOp, error
 			switch spec.Tp {
 			case ast.PlacementAdd:
 				rule.Action = placement.RuleOpAdd
-			case ast.PlacementAlter:
+			case ast.PlacementAlter, ast.PlacementDrop:
 				rule.Action = placement.RuleOpAdd
```
Comment (tangenta) [inline ddl/ddl_api.go:5256]: When do we use `RuleOpDel`? Why do we need this `switch` snippet?

Comment (xhebox) [reply ddl/ddl_api.go:5256]: `RuleOpDel` is at L5272. DDL stmts are not that closely related to PD api. Alter need to make use of both `RuleOpAdd` and `RuleOpDel`. This line is prepared for later half part - placement rules building. But drop will use continue to skip building and use `RuleOpDel` only.

Switch is used to make things more clear initially. But, yes, now it looks like a nested if-else statement should do the work, too.
It is unexpected that drop and alter can share same code.
I am OK on both switch and if-else, do I need to change to if-else?

**Pull request merged**